### PR TITLE
Adding button

### PR DESCRIPTION
**Pull request by gabrielreisn**
This PR attempts to add the button component and basic configuration for CI / CD

### CODE REVIEW

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+
+type buttonProps = {
```
Comment (savioserra) [inline src/components/Button/index.tsx:4]: Type name should be PascalCase.
```suggestion
type ButtonProps = {
```

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  lg: '20px 40px',
+};
+
+const fonts = {
+  sm: '10px',
+  md: '15px',
+  lg: '25px',
+};
+
+const Button = styled.button<buttonProps>`
```
Comment (savioserra) [inline src/components/Button/index.tsx:21]:
```suggestion
const Button = styled.button<ButtonProps>`
```

**Review by savioserra — CHANGES_REQUESTED** [review]
Everything's ok except the type name, which should be PascalCase 😺 
Good work! 😃

**Comment by gabrielreisn** [comment]
@savioserra it's on WIP, so don't worry about reviewing it :) Besides the PascalCase it needs to have more themes than just "tomato" and "disabled"

**Comment by savioserra** [comment]
We should establish a color palette in our guideline.

Also, `emotionjs` has a component named `ThemeProvider`, which injects themes as props on child components. When I was developing the InputField component (also wip), I created a `default` folder inside the `themes` folder. There, we can define those colors and others variables, which should fallback to when the user doesn't provide his own theme. 

image

What do you think? I think this is a nice approach for both library and users. We could split the code inside the `themes/default` as: `themes/default/colors`, `themes/default/sizes?` and so on, and export it all together in the `themes/default/index.tsx`.

**Comment by savioserra** [comment]
About the prop type. Do you think we should keep it as a separate file or define it in the same file of the component?

**Comment by gabrielreisn** [comment]
@savioserra ok, i'll try to use and improve the emotion theme provider. We can discuss color pallet at the guidelines. WDYT about using a movie / anime for choosing it? (like this: https://digitalsynopsis.com/wp-content/uploads/2016/05/cinema-palettes-famous-movie-colors-moonrise-kingdom.jpg)

I also think that we should keep consistency and add a `types` file for each component, just like you did :)

**Comment by savioserra** [comment]
@gabrielreisn What if we wrap our top level components always on a `ThemeProvider`, like this:
```javascript
<ThemeProvider theme={props.theme || defaultTheme} >
    {...children}
<ThemeProvider/>
```

Or even better, we could **merge** them:
```javascript
<ThemeProvider theme={{...defaultTheme, ...props.theme}} >
    {...children}
<ThemeProvider/>
```

So we don't need to check every time if a value is present in the `props.theme`, since it always falls back to the `defaultTheme`.

**Comment by savioserra** [comment]
> @savioserra ok, i'll try to use and improve the emotion theme provider. We can discuss color pallet at the guidelines. WDYT about using a movie / anime for choosing it? (like this: https://digitalsynopsis.com/wp-content/uploads/2016/05/cinema-palettes-famous-movie-colors-moonrise-kingdom.jpg)
> 
> I also think that we should keep consistency and add a `types` file for each component, just like you did :)

Choosing the color palette based on a anime sounds funny 😃 Let's do it.

**Pull request merged**